crates/adapters/src/transport/s3/mod.rs

```diff
@@ -477,25 +488,34 @@ impl<T> BufferedReceiver<T> {
     }
 }
 
-fn to_s3_config(config: &Arc<S3InputConfig>) -> aws_sdk_s3::Config {
-    let config_builder = if let Some(endpoint) = &config.endpoint_url {
-        aws_sdk_s3::Config::builder()
-            .region(aws_types::region::Region::new(config.region.clone()))
-            .endpoint_url(endpoint)
-    } else {
-        aws_sdk_s3::Config::builder().region(aws_types::region::Region::new(config.region.clone()))
-    };
+fn to_s3_config(config: &Arc<S3InputConfig>, irsa: bool) -> aws_sdk_s3::Config {
+    let mut config_builder =
+        aws_sdk_s3::Config::builder().region(aws_types::region::Region::new(config.region.clone()));
+
+    if let Some(endpoint) = &config.endpoint_url {
+        config_builder = config_builder.endpoint_url(endpoint);
+    }
+
     if config.no_sign_request {
-        config_builder.build()
-    } else {
+        return config_builder.build();
+    }
+
+    if !irsa {
```

Comment:
Instead of passing this bool throughout the function signatures, you can just check in this line that config.aws_access_key_id and config.aws_secret_access_key was supplied.

Reply:
Yes but we already check for it in the validation logic in `S3InputEndpoint::new`. Opted against a redundant check here.

But its a pretty tiny check, sure, I will just do it here.